src/main/java/ch/njol/skript/expressions/ExprAdvancementsOfPlayer.java

```diff
@@ -0,0 +1,152 @@
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("Player Advancements")
+@Description("The advancements of a player.")
+@Examples({
+	"add the advancement \"minecraft:adventure/root\" to advancements of player",
+	"if advancements of player contains the advancement \"minecraft:adventure/root\"",
+	"remove all advancements from advancements of player"
+})
+@Since("INSERT VERSION")
+public class ExprAdvancementsOfPlayer extends SimpleExpression<Advancement> {
+
+	static {
+		Skript.registerExpression(ExprAdvancementsOfPlayer.class, Advancement.class, ExpressionType.SIMPLE,
+			"[all] [[of] the] advancements (of|from) %players%",
+			"[all [of]] %players%'[s] advancements");
+	}
+
+	private Expression<Player> players;
+	private boolean single;
+
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return "the advancements of " + players.toString(e, debug);
+	}
+
+	@Override
+	public boolean isSingle() {
+		return single;
+	}
+
+	@Override
+	public Class<? extends Advancement> getReturnType() {
+		return Advancement.class;
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		players = (Expression<Player>) exprs[0];
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case ADD:
+			case REMOVE:
+			case DELETE:
+			case RESET:
+				return CollectionUtils.array(Advancement[].class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		List<Advancement> advancements = new ArrayList<>();
+		for (Player player : players.getArray(e)) {
+			switch (mode) {
+				case SET:
+					assert delta != null;
+					advancements.addAll(Arrays.asList((Advancement[]) delta));
+					break;
+				case ADD:
+					assert delta != null;
+					advancements.addAll(List.of(getAdvancementsFromPlayer(player)));
+					advancements.addAll(Arrays.asList((Advancement[]) delta));
+					break;
+				case REMOVE:
+					assert delta != null;
+					advancements.addAll(List.of(getAdvancementsFromPlayer(player)));
+					advancements.removeAll(Arrays.asList((Advancement[]) delta));
+					break;
+				case DELETE:
+				case RESET:
+					break;
+			}
+			AdvancementProgress progress;
+			for (Advancement advancement : getAdvancementsFromPlayer(player)) {
+				progress = player.getAdvancementProgress(advancement);
+				for (String criteria : progress.getAwardedCriteria())
+					progress.revokeCriteria(criteria);
+			}
+			for (Advancement advancement : advancements) {
+				progress = player.getAdvancementProgress(advancement);
+				for (String criteria : progress.getRemainingCriteria())
+					progress.awardCriteria(criteria);
+			}
+		}
+	}
+
+	@Override
+	protected @Nullable Advancement[] get(Event e) {
```

Comment:
`event` instead of `e`
and Nullable should go on a new line above after Override